File: server/src/services/templates.ts
import { type ProjectTemplate } from '@scribe/shared';

interface TemplateFile {
  readonly path: string;
  readonly content: string;
}

const BLANK: readonly TemplateFile[] = [{ path: 'main.tex', content: '' }];

const ARTICLE: readonly TemplateFile[] = [
  {
    path: 'main.tex',
    content: `\\documentclass[11pt]{article}
\\usepackage[utf8]{inputenc}
\\usepackage{amsmath, amssymb, amsthm}
\\usepackage{graphicx}
\\usepackage{hyperref}

\\title{Untitled}
\\author{}
\\date{\\today}

\\begin{document}
\\maketitle

\\section{Introduction}
Welcome to Scribe.

\\end{document}
`,
  },
];

const REPORT: readonly TemplateFile[] = [
  {
    path: 'main.tex',
    content: `\\documentclass[11pt]{report}
\\usepackage[utf8]{inputenc}

\\title{Untitled Report}
\\author{}
\\date{\\today}

\\begin{document}
\\maketitle
\\tableofcontents

\\chapter{Introduction}

\\end{document}
`,
  },
];

const BEAMER: readonly TemplateFile[] = [
  {
    path: 'main.tex',
    content: `\\documentclass{beamer}
\\usetheme{metropolis}

\\title{Untitled}
\\author{}
\\date{\\today}

\\begin{document}
\\maketitle

\\begin{frame}{Hello}
  Welcome to Scribe.
\\end{frame}

\\end{document}
`,
  },
];

const CV: readonly TemplateFile[] = [
  {
    path: 'main.tex',
    content: `\\documentclass[11pt,a4paper]{moderncv}
\\moderncvstyle{classic}
\\moderncvcolor{blue}

\\name{Your}{Name}
\\title{Title}

\\begin{document}
\\makecvtitle

\\section{Experience}
\\cventry{2024--Present}{Role}{Company}{Location}{}{Description.}

\\end{document}
`,
  },
];

const LETTER: readonly TemplateFile[] = [
  {
    path: 'main.tex',
    content: `\\documentclass{letter}
\\signature{Your Name}
\\address{Your Address}

\\begin{document}
\\begin{letter}{Recipient Name \\\\ Recipient Address}
\\opening{Dear Recipient,}

Body goes here.

\\closing{Sincerely,}
\\end{letter}
\\end{document}
`,
  },
];

const TEMPLATES: Readonly<Record<ProjectTemplate, readonly TemplateFile[]>> = {
  blank: BLANK,
  article: ARTICLE,
  report: REPORT,
  beamer: BEAMER,
  cv: CV,
  letter: LETTER,
};

export function filesForTemplate(template: ProjectTemplate): readonly TemplateFile[] {
  return TEMPLATES[template];
}

File: servers/rust/crates/scribe-server/src/services/templates.rs
//! Per-template seed files. Mirrors `server/src/services/templates.ts`
//! verbatim so new projects produced by either server are identical.

use scribe_shared::ProjectTemplate;

pub struct TemplateFile {
    pub path: &'static str,
    pub content: &'static str,
}

const BLANK: &[TemplateFile] = &[TemplateFile { path: "main.tex", content: "" }];

const ARTICLE: &[TemplateFile] = &[TemplateFile {
    path: "main.tex",
    content: "\\documentclass[11pt]{article}\n\
\\usepackage[utf8]{inputenc}\n\
\\usepackage{amsmath, amssymb, amsthm}\n\
\\usepackage{graphicx}\n\
\\usepackage{hyperref}\n\
\n\
\\title{Untitled}\n\
\\author{}\n\
\\date{\\today}\n\
\n\
\\begin{document}\n\
\\maketitle\n\
\n\
\\section{Introduction}\n\
Welcome to Scribe.\n\
\n\
\\end{document}\n",
}];

const REPORT: &[TemplateFile] = &[TemplateFile {
    path: "main.tex",
    content: "\\documentclass[11pt]{report}\n\
\\usepackage[utf8]{inputenc}\n\
\n\
\\title{Untitled Report}\n\
\\author{}\n\
\\date{\\today}\n\
\n\
\\begin{document}\n\
\\maketitle\n\
\\tableofcontents\n\
\n\
\\chapter{Introduction}\n\
\n\
\\end{document}\n",
}];

const BEAMER: &[TemplateFile] = &[TemplateFile {
    path: "main.tex",
    content: "\\documentclass{beamer}\n\
\\usetheme{metropolis}\n\
\n\
\\title{Untitled}\n\
\\author{}\n\
\\date{\\today}\n\
\n\
\\begin{document}\n\
\\maketitle\n\
\n\
\\begin{frame}{Hello}\n\
  Welcome to Scribe.\n\
\\end{frame}\n\
\n\
\\end{document}\n",
}];

const CV: &[TemplateFile] = &[TemplateFile {
    path: "main.tex",
    content: "\\documentclass[11pt,a4paper]{moderncv}\n\
\\moderncvstyle{classic}\n\
\\moderncvcolor{blue}\n\
\n\
\\name{Your}{Name}\n\
\\title{Title}\n\
\n\
\\begin{document}\n\
\\makecvtitle\n\
\n\
\\section{Experience}\n\
\\cventry{2024--Present}{Role}{Company}{Location}{}{Description.}\n\
\n\
\\end{document}\n",
}];

const LETTER: &[TemplateFile] = &[TemplateFile {
    path: "main.tex",
    content: "\\documentclass{letter}\n\
\\signature{Your Name}\n\
\\address{Your Address}\n\
\n\
\\begin{document}\n\
\\begin{letter}{Recipient Name \\\\ Recipient Address}\n\
\\opening{Dear Recipient,}\n\
\n\
Body goes here.\n\
\n\
\\closing{Sincerely,}\n\
\\end{letter}\n\
\\end{document}\n",
}];

pub fn files_for_template(template: ProjectTemplate) -> &'static [TemplateFile] {
    match template {
        ProjectTemplate::Blank => BLANK,
        ProjectTemplate::Article => ARTICLE,
        ProjectTemplate::Report => REPORT,
        ProjectTemplate::Beamer => BEAMER,
        ProjectTemplate::Cv => CV,
        ProjectTemplate::Letter => LETTER,
    }
}
